DbMocker.Tests/ConversionTests.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Apps72.Dev.Data.DbMocker;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Linq;
+
+namespace DbMocker.Tests
+{
+    [TestClass]
+    public class ConversionTests
+    {
+        [TestMethod]
+        public void Mock_EmptyTable_Test()
+        {
+            var mockTable = MockTableExtensions.GetEmptyMockTable<RowDto>();
+
+            Assert.AreEqual(0, mockTable.Rows.Length);
+            Assert.AreEqual(4, mockTable.Columns.Length);
+        }
+
+        [TestMethod]
+        public void Mock_EmptyTable_Test2()
+        {
+            var dto = (Id: 1, Name: "Mock", Price: (decimal?)null);
+            var mockTable = dto.GetEmptyMockTable((s) => s.Add(k => k.Name, nameof(dto.Name)));
+            Assert.AreEqual(0, mockTable.Rows.GetLength(0));
+            Assert.AreEqual(1, mockTable.Columns.Length);
+            Assert.IsTrue(mockTable.Columns.Any(a => a.Name.Equals("Name")));
+        }
+
+        [TestMethod]
+        public void Mock_EmptyTable_SelectedColumn_Test()
+        {
+            var mockTable = MockTableExtensions.GetEmptyMockTable<RowDto>(f => f.Add(r => r.Id).Add(r => r.Price));
+
+            Assert.AreEqual(0, mockTable.Rows.Length);
+            Assert.AreEqual(2, mockTable.Columns.Length);
+            Assert.IsFalse(mockTable.Columns.Any(a => a.Name.Equals(nameof(RowDto.Name))));
+        }
+
+        [TestMethod]
+        public void Mock_ObjectConversion_Test()
+        {
+            var dto = new RowDto() { Id = 1, Name = "Mock", Price = null };
+            var mockTable = dto.ToMockTable();
+            Assert.AreEqual(1, mockTable.Rows.GetLength(0));
+            Assert.AreEqual(4, mockTable.Columns.Length);
+        }
+
+        [TestMethod]
+        public void Mock_NamedTupleConversion_Test()
+        {
+            var dto = (Id: 1, Name: "Mock", Price: (decimal?)null);
+            var mockTable = dto.ToMockTable("ID", "MaimName", "Price");
+            Assert.AreEqual(1, mockTable.Rows.GetLength(0));
+            Assert.AreEqual(3, mockTable.Columns.Length);
+            Assert.IsTrue(mockTable.Columns.Any(a => a.Name.Equals("MaimName")));
+        }
+
+        [TestMethod]
+        public void Mock_NamedTupleConversion_Test2()
+        {
+            var dto = (Id: 1, Name: "Mock", Price: (decimal?)null);
+            var mockTable = dto.ToMockTable((s) => s.Add(k => k.Name, "MaimName"));
```

Comment:
I found this syntax a little complex. Do you think it's possible to simplify?

Reply:
It is impossible. We will lose control.